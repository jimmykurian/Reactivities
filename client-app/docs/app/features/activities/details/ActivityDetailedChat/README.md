[**Client App Documentation v0.0.0**](../../../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../../../README.md) / app/features/activities/details/ActivityDetailedChat

# app/features/activities/details/ActivityDetailedChat

## Index

### Functions

- [default](functions/default.md)
