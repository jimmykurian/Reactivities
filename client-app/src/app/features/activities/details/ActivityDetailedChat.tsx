/**
 * @file ActivityDetailedChat.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the ActivityDetailedChat component that displays the chat section for an activity's details page.
 */

import { observer } from 'mobx-react-lite';
import { Segment, Header, Comment, Form, Button } from 'semantic-ui-react';

/**
 * @component ActivityDetailedChat
 * @description The ActivityDetailedChat component renders a chat interface for users to discuss the event. It includes a list of comments and a form to add new replies.
 *
 * @returns {JSX.Element} The rendered ActivityDetailedChat component.
 *
 * @remarks
 * This component uses Semantic UI React components to create a chat interface. It displays a header, a list of comments, and a form for adding new replies. Each comment includes an avatar, author, metadata, text, and actions for replying.
 *
 * @example
 * Here is an example of how to use the ActivityDetailedChat component:
 * ```tsx
 * import React from 'react';
 * import ActivityDetailedChat from './features/activities/details/ActivityDetailedChat';
 *
 * const App = () => (
 *   <div>
 *     <ActivityDetailedChat />
 *   </div>
 * );
 *
 * export default App;
 * ```
 */
export default observer(function ActivityDetailedChat(): JSX.Element {
  return (
    <>
      <Segment
        textAlign="center"
        attached="top"
        inverted
        color="teal"
        style={{ border: 'none' }}
      >
        <Header>Chat about this event</Header>
      </Segment>
      <Segment attached>
        <Comment.Group>
          <Comment>
            <Comment.Avatar src="/assets/user.png" />
            <Comment.Content>
              <Comment.Author as="a">Matt</Comment.Author>
              <Comment.Metadata>
                <div>Today at 5:42PM</div>
              </Comment.Metadata>
              <Comment.Text>How artistic!</Comment.Text>
              <Comment.Actions>
                <Comment.Action>Reply</Comment.Action>
              </Comment.Actions>
            </Comment.Content>
          </Comment>

          <Comment>
            <Comment.Avatar src="/assets/user.png" />
            <Comment.Content>
              <Comment.Author as="a">Joe Henderson</Comment.Author>
              <Comment.Metadata>
                <div>5 days ago</div>
              </Comment.Metadata>
              <Comment.Text>Dude, this is awesome. Thanks so much</Comment.Text>
              <Comment.Actions>
                <Comment.Action>Reply</Comment.Action>
              </Comment.Actions>
            </Comment.Content>
          </Comment>

          <Form reply>
            <Form.TextArea />
            <Button
              content="Add Reply"
              labelPosition="left"
              icon="edit"
              primary
            />
          </Form>
        </Comment.Group>
      </Segment>
    </>
  );
});
