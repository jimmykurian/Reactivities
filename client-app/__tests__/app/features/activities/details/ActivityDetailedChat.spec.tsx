import { render, screen, fireEvent } from '@testing-library/react';
import ActivityDetailedChat from '../../../../../src/app/features/activities/details/ActivityDetailedChat';

// Mocking observer from mobx-react-lite
jest.mock('mobx-react-lite', () => ({
  observer: (component: React.ComponentType) => component,
}));

describe('ActivityDetailedChat', () => {
  // Static mock data for snapshot test
  const staticComments = [
    {
      avatar: '/assets/user.png',
      author: 'Matt',
      metadata: 'Today at 5:42PM',
      text: 'How artistic!',
    },
    {
      avatar: '/assets/user.png',
      author: 'Joe Henderson',
      metadata: '5 days ago',
      text: 'Dude, this is awesome. Thanks so much',
    },
  ];

  test('renders ActivityDetailedChat component', () => {
    // Arrange & Act
    render(<ActivityDetailedChat />);

    // Assert
    staticComments.forEach((comment) => {
      expect(screen.getByText(comment.author)).toBeInTheDocument();
      expect(screen.getByText(comment.metadata)).toBeInTheDocument();
      expect(screen.getByText(comment.text)).toBeInTheDocument();
    });
    expect(screen.getByText('Chat about this event')).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /Add Reply/i }),
    ).toBeInTheDocument();
  });

  test('allows user to type in the reply form', () => {
    // Arrange
    render(<ActivityDetailedChat />);
    const textarea = screen.getByRole('textbox');

    // Act
    fireEvent.change(textarea, { target: { value: 'New reply' } });

    // Assert
    expect(textarea).toHaveValue('New reply');
  });

  test('matches snapshot', () => {
    // Arrange & Act
    const { asFragment } = render(<ActivityDetailedChat />);

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
