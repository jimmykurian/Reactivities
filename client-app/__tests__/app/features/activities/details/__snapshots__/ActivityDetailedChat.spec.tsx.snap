// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDetailedChat matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui teal inverted top attached center aligned segment"
  >
    <div
      class="ui header"
    >
      Chat about this event
    </div>
  </div>
  <div
    class="ui attached segment"
  >
    <div
      class="ui comments"
    >
      <div
        class="comment"
      >
        <div
          class="avatar"
        >
          <img
            src="/assets/user.png"
          />
        </div>
        <div
          class="content"
        >
          <a
            class="author"
          >
            Matt
          </a>
          <div
            class="metadata"
          >
            <div>
              Today at 5:42PM
            </div>
          </div>
          <div
            class="text"
          >
            How artistic!
          </div>
          <div
            class="actions"
          >
            <a
              class=""
            >
              Reply
            </a>
          </div>
        </div>
      </div>
      <div
        class="comment"
      >
        <div
          class="avatar"
        >
          <img
            src="/assets/user.png"
          />
        </div>
        <div
          class="content"
        >
          <a
            class="author"
          >
            Joe Henderson
          </a>
          <div
            class="metadata"
          >
            <div>
              5 days ago
            </div>
          </div>
          <div
            class="text"
          >
            Dude, this is awesome. Thanks so much
          </div>
          <div
            class="actions"
          >
            <a
              class=""
            >
              Reply
            </a>
          </div>
        </div>
      </div>
      <form
        class="ui reply form"
      >
        <div
          class="field"
        >
          <textarea
            rows="3"
          />
        </div>
        <button
          class="ui icon primary left labeled button"
        >
          <i
            aria-hidden="true"
            class="edit icon"
          />
          Add Reply
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
